### PR TITLE
Add contribution guidelines to the documentation

### PR DESCRIPTION
**Pull request by bwroblew**

### CODE REVIEW

File: docs/Contributors_Guide.rst [hunk 1]
```diff
@@ -2,7 +2,96 @@
...
+
+When submitting a Pull Request you should:
+
+* Describe the change providing information about the motivation for the change and a general
+  description of all code modifications.
+
+* Verify and test the change:
+
+  * Run any relevant existing tests.
+  * Write new tests if the existing ones do not exist or do not cover the functionality entirely.
```
Comment (aosewski) [inline docs/Contributors_Guide.rst]:
```suggestion
  * Write new tests if added functionality is not covered by current tests.
```

File: docs/Contributors_Guide.rst [hunk 2]
```diff
@@ -2,7 +2,96 @@
...
+
+* Link your PR to any related issues:
+
+  * If there is an issue that is resolved by your change, please provide a link to the issue in
+    the description of your pull request.
+  * If there is no issue related to your change, we recommend creating a new issue and then
```
Comment (aosewski) [inline docs/Contributors_Guide.rst]: This point seems to me ambiguous. Not every PR is fixing some issues.

File: docs/Contributors_Guide.rst [hunk 3]
```diff
@@ -2,7 +2,96 @@
...
+Before sumbitting any issue, ensure you have addressed all relevant questions from the checklist.
+
+Creating Pull Requests
+----------------------
+
+You can submit `Pull Requests (PR) on GitHub
+<https://github.com/ROCmSoftwarePlatform/composable_kernel/pulls>`_.
+
+All contributors are required to develop their changes on a separate branch and then create a
+pull requrest to merge their changes into the `develop` branch, which is the default
```
Comment (aosewski) [inline docs/Contributors_Guide.rst:71]: Isn't like users outside of ROCmSoftwarePlatform have to create their own fork of CK and can create PR only from their fork to our repo ?

**Review by aosewski — COMMENTED** [review]
Overall looks great. Added few minor comments. 

You could also point out the need of adhering to the specific code-formatting rules we use.

File: docs/Contributors_Guide.rst (hunk 2 above)
Comment (bwroblew) [reply docs/Contributors_Guide.rst]: removed

File: docs/Contributors_Guide.rst (hunk 3 above)
Comment (bwroblew) [reply docs/Contributors_Guide.rst:71]: right, I added another sentence about the need for using forks

**Comment by bwroblew** [comment]
> You could also point out the need of adhering to the specific code-formatting rules we use.

Sure, forgot about that. Added a new bullet point about using `clang-format` and `pre-commit`.

**Review by illsilin — APPROVED** [review]

**Pull request merged**